### PR TITLE
Misc. Cleanups

### PR DESCRIPTION
**Pull request by nbougalis**

### CODE REVIEW

**Comment by MarkusTeufelberger** [comment]
While you're at it:

https://github.com/ripple/rippled/blob/develop/src/ripple/unity/hyperleveldb.h
https://github.com/ripple/rippled/blob/develop/src/ripple/unity/leveldb.h

File: src/ripple/app/tx/impl/CancelTicket.cpp [hunk 1]
```diff
@@ -32,6 +32,9 @@ CancelTicket::preflight (PreflightContext const& ctx)
     if (! ctx.rules.enabled(featureTickets))
         return temDISABLED;
 
+    if (! ctx.rules.enabled(fix1543) && ctx.tx.getFlags() & tfUniversalMask)
```
Comment (seelabs) [inline src/ripple/app/tx/impl/CancelTicket.cpp]: This condition doesn't look right. We want the extra check if `fix1543` is enabled, not disabled.

**Comment by seelabs** [comment]
`CancelTicket.cpp` and `CreateTicket.cpp` need to include `#include <ripple/protocol/TxFlags.h>` for the non-unity build.

File: src/ripple/app/tx/impl/CancelTicket.cpp (hunk 1 above)
Comment (nbougalis) [reply src/ripple/app/tx/impl/CancelTicket.cpp]: I meant to remove the amendment check entirely, actually. `CreateTicket` and `CancelTicket` aren't enabled yet, so we can modify their semantics directly.

Comment (seelabs) [reply src/ripple/app/tx/impl/CancelTicket.cpp]: Cool, although Escrow and PayChan have the same backward condition and need to be fixed.

Comment (nbougalis) [reply src/ripple/app/tx/impl/CancelTicket.cpp]: Yep, done.

File: src/ripple/app/tx/impl/PayChan.cpp [hunk 2]
```diff
@@ -348,6 +354,9 @@ PayChanClaim::preflight (PreflightContext const& ctx)
     if (! ctx.rules.enabled(featurePayChan))
         return temDISABLED;
 
+    if (ctx.rules.enabled(fix1543) && ctx.tx.getFlags() & tfUniversalMask)
+        return temINVALID_FLAG;
```
Comment (seelabs) [inline src/ripple/app/tx/impl/PayChan.cpp]: This should be removed - claim can take flags. See line 383.

Comment (nbougalis) [reply src/ripple/app/tx/impl/PayChan.cpp]: Then we need to define those flags. I'll upgrade.

File: src/ripple/beast/unity/beast_hash_unity.cpp [hunk 3]
```diff
@@ -18,9 +18,7 @@
 //==============================================================================
 
 #include <ripple/beast/hash/xxhasher.h>
-#include <ripple/beast/hash/impl/spookyv2.cpp>
 
 #if ! BEAST_NO_XXHASH
```
Comment (mellery451) [inline src/ripple/beast/unity/beast_hash_unity.cpp]: I think `BEAST_NO_XXHASH` might be unsupported now?  I tried building with it set and got several errors, so I think it's not an option any more.

Comment (nbougalis) [reply src/ripple/beast/unity/beast_hash_unity.cpp]: Good catch!

**Comment by MarkusTeufelberger** [comment]
https://github.com/ripple/rippled/blob/develop/src/ripple/unity/beastobjc.mm
https://github.com/ripple/rippled/blob/develop/src/ripple/beast/core/mac_SystemStats.mm
https://github.com/ripple/rippled/blob/develop/src/ripple/beast/core/core.unity.mm

**Comment by MarkusTeufelberger** [comment]
https://github.com/ripple/rippled/blob/35fa20a110e3d43ffc1e9e664fc9017b6f2747ae/src/ripple/protocol/digest.h#L125-L133

Replacing hash functions from OpenSSL? These are hopefully as fast and standard as can be already...

Also OpenSSL and BoringSSL (probably LibreSSL too, haven't checked there) now also natively support ed25519, so you might be able to drop another whole library from your code base. :-)

**Comment by nbougalis** [comment]
We compile _with_ `RIPPLE_USE_OPENSSL`, so we use the OpenSSL versions. But Beast had its own implementations (for other purposes) so it was easy to offer a common interface for both. Hence that block of code.

It's cool that OpenSSL has Ed25519 support, but I ran a quick check and it seems horribly slow compared to the current library; and, honestly, I find the API that it exposes to be horrible.

**Comment by MarkusTeufelberger** [comment]
So the else part is already dead code and can't even compile. `RIPPLE_USE_OPENSSL` is only used in this single place in the code, so it probably can be removed alltogether.

**Comment by nbougalis** [comment]
@MarkusTeufelberger: removing those files will take a bit more effort. It's on the queue to happen.

**Pull request closed without merging**